hw/dv/tools/sim.tcl

```diff
@@ -38,9 +38,17 @@ if {$simulator eq "xcelium"} {
   assertion -list -depth all -multiline -permoff $tb_top
 }
 
-# In GUI mode, let the user take control of running the simulation.
 global gui
-if {$gui == 0} {
+global gui_debug
+# In GUI mode, let the user take control of running the simulation.
+# In GUI debug mode, run only up until the end of the UVM elaboration phase to have access to the
+# dynamic objects.
+if {$gui_debug == 1} {
+  if {$simulator eq "xcelium"} {
+    uvm_phase -stop_at build -end
+    run
+  }
+} elseif {$gui == 0 && $gui_debug == 0} {
```

Comment:
I'd probably drop the `gui_debug` check here. At the moment, it's correct (but unnecessary) provided that `$gui_debug` is always 0 or 1. I'm not sure it does something sensible if `gui_debug` is 2...

If you drop that check here, we end up treating `gui_debug=cabbage` as `gui_debug=0`. Alternatively, we could change line 46 to be `$gui_debug != 0` (so treating silly values as 1) or add a check to make sure the value is something sensible.

Reply:
Yes you're right, I can drop it. I think `gui_debug` cannot be anything else than `0/1` as this is defined in this script `hw/dv/tools/common.tcl` and not by the user himself.